http/api.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+
+	. "github.com/StevensSEC/pokemonbattlelib"
+)
+
+func buildRouter() *http.ServeMux {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/pokedex/list/moves", HandleListMoves)
+	mux.HandleFunc("/pokedex/generate", HandleGeneratePokemon)
+	mux.HandleFunc("/battle/new", HandleCreateBattle)
+	mux.HandleFunc("/battle/simulate", HandleBattleSimulate)
+	mux.HandleFunc("/agent/dumb", HandleDumbAgent)
+	return mux
+}
+
+// List all available moves.
+func HandleListMoves(w http.ResponseWriter, r *http.Request) {
+	bytes, err := json.Marshal(ALL_MOVES)
+	if err != nil {
+		log.Fatalf("Failed to marshal moves into JSON: %s", err)
+	}
+	w.WriteHeader(200)
+	w.Write(bytes)
+}
+
+type httpPokemon struct {
+	Pokemon
+	Name string
+}
+
+// Convert a Pokemon into an httpPokemon, suitable for JSON marshalling for HTTP responses that contain pokemon.
+func convertPokemon(p *Pokemon) *httpPokemon {
+	return &httpPokemon{
+		Pokemon: *p,
+		Name:    p.GetName(),
+	}
+}
```

Comment:
This kind of pattern should be avoided IMO. Add JSON tags to relevant structs [shown here](https://gobyexample.com/json).

Reply:
But we need the name, which is not present on `Pokemon`.

Reply:
You can write a custom `MarshalJSON` and `UnmarshalJSON` method on the original struct which can have custom effects. See [this article](http://choly.ca/post/go-json-marshalling/) for a simple example.

Reply:
Ah, I didn't think to introduce another struct to marshal the data

Reply:
Ok, so it turns out that you can't implement methods on types that you don't define, and since `Pokemon` is not defined in this package, I can't implement `MarshalJSON`/`UnmarshalJSON`

Reply:
I don't see any problem with defining those methods in the original library, even if it's in a separate file/in `util.go`. Can even make it take a function so that it's customizable such as `p.MarshalJSON(customHandlerFunc)`. Also, have you looked at [WASM](https://www.sitepen.com/blog/compiling-go-to-webassembly)?

Reply:
I don't see the relavance of web assembly, just seems like extra steps.

Reply:
Also, if we change the function signature of `MarshalJSON` then we don't implement the `Marshaler` interface.